### ISSUE TITLE
Some bugs with Warship and Book of Changes

### ISSUE THREAD

**Issue by BabyGrootTheEpic** [fantasy-realms/fantasy-realms.github.io#6]
I found 3 bugs that can happen with Warship when Book of Changes changes something to or from a Flood:

- If a card is turned into a Flood by Book of Changes, Army is not cleared from its penalty.
[Basilisk blanks Knights (an Army) despite being a Flood.](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR37,FR21+FR49:FR37:flood)
[Dwarvish Infantry's is penalized for being with Elven Archers (an Army) despite being a Flood.](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR24,FR22+FR49:FR24:flood)

- If War Drigible is changed into a Flood, Warship will see the Flood and won't be blanked even if War Drigible is blanked by its penalty.
[Blanked by lack of Army](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR45+FR49:FR45:flood)
[Blanked by Weather](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR45,FR23,FR15+FR49:FR45:flood)

- If Swamp or Great Flood are changed into another suit by Book of Changes, Warship will still clear "Army" from their penalty.
[Swamp example](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR10,FR07,FR22+FR49:FR07:wizard)
[Great Flood example](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR06,FR08,FR22+FR49:FR08:wizard)

**Comment by BabyGrootTheEpic** [comment]
Just found a 4th bug with Warship (it doesn't require Book of Changes though):
- If Warship is blanked, it will not clear the word "Army" from a flood's penalty.
[Example with Swamp](https://fantasy-realms.github.io/index.html?expansions=ch_suits&playerCount=4&hand=CH10,FR41,FR10,FR07,FR22,FR23+)
[Example with Great Flood](https://fantasy-realms.github.io/index.html?expansions=ch_suits&playerCount=4&hand=CH10,FR41,CH18,FR10,FR22,FR23+)

**Comment by jpraet** [comment]
First and third issue should be fixed.

**Comment by jpraet** [comment]
4th issue is also fixed.

I am not sure what should happen for the 2nd issue. It seems like this scenario is not clearly covered by the rules.

For [Blanked by lack of Army](https://fantasy-realms.github.io/index.html?hand=FR49,FR41,FR45+FR49:FR45:flood) one could even argue that War Dirigible should not get blanked? What does it mean to clear the word "Army" from "BLANKED unless with at least one Army."?

**Comment by BabyGrootTheEpic** [comment]
Yeah I don't know how clearing "Army" from War Drigible would work, but either way in the second issue I was referencing how Warship was seing a blanked flood as the card to unblank it.

**Comment by jpraet** [comment]
I'm not sure how "BLANKED with X" and "BLANKED unless with at least one X" are supposed to interact.

Consider:
* War Dirigible (Weapon): BLANKED with any Weather
* Smoke (Weather): BLANKED unless with at least one Flame
* Use Book of Changes to change War Dirigible into a Flame

Smoke causes War Dirigible to be blanked, which causes Smoke to be blanked, which causes War Dirigible to no longer be blanked, which causes Smoke to no longer be blanked, which causes War Dirigible to be blanked, which...
